harness/determined/pytorch/_trainer.py

```diff
@@ -0,0 +1,211 @@
+
+    def configure_profiler(
+        self, sync_timings: bool, enabled: bool, begin_on_batch: int, end_after_batch: int
+    ):
+        assert self._cluster_info, "Determined profiler must be run on cluster"
+        self._det_profiler = ProfilerAgent(
+            trial_id=str(self._cluster_info.trial.trial_id),
+            agent_id=self._cluster_info.agent_id,
+            master_url=self._cluster_info.master_url,
+            profiling_is_enabled=enabled,
+            global_rank=self._core.distributed.get_rank(),
+            local_rank=self._core.distributed.get_local_rank(),
+            begin_on_batch=begin_on_batch,
+            end_after_batch=end_after_batch,
+            sync_timings=sync_timings,
+        )
+
+    def train(
+        self,
+        max_epochs: Optional[int] = None,
+        # OR
+        max_batches: Optional[int] = None,
+        min_checkpoint_period: Union[pytorch.TrainUnit, int] = 1,
+        min_validation_period: Union[pytorch.TrainUnit, int] = 1,
+        average_training_metrics: Optional[bool] = True,
+        average_aggregated_gradients: Optional[bool] = True,
+        aggregation_frequency: Optional[int] = 1,
+        checkpoint_policy: Optional[str] = "best",
+        smaller_is_better: Optional[bool] = True,
+    ):
+
+        if self._local_training:
+            assert (max_epochs is None) ^ (
+                max_batches is None
+            ), "Either max_batches or max_epochs must be defined in local training mode"
+        else:
+            if max_batches or max_epochs:
+                logging.warning(
+                    "max_batches and max_epochs is ignored in when training on cluster. "
+                    "Please configure the searcher length instead."
+                )
+
+        # Set context and training variables
+        self._context._aggregation_frequency = aggregation_frequency
+        self._context._average_aggregated_gradients = average_aggregated_gradients
+
+        max_length = None
+        if max_batches:
+            max_length = pytorch.Batch(max_batches)
+        elif max_epochs:
+            max_length = pytorch.Epoch(max_epochs)
+
+        # Convert validation/checkpoint periods to training units.
+        # Without a specified training unit, periods will be assumed to be same as max_length in local training mode,
+        # and the searcher unit when training on-cluster
+        if isinstance(min_checkpoint_period, int):
+            min_checkpoint_period = self._convert_period_to_train_unit(
+                min_checkpoint_period, max_length
+            )
+
+        if isinstance(min_validation_period, int):
+            min_validation_period = self._convert_period_to_train_unit(
+                min_validation_period, max_length
+            )
+
+        if self._local_training:
+            if checkpoint_policy != "all":
+                logging.warning("Checkpoint policy set to 'all' in local training mode.")
```

Comment:
why is this necessary?

Reply:
because local training doesn't have the concept of a searcher, which we rely on to determine the "best" checkpoint

Reply:
I think since we believe that `"best"` is a useful value on the cluster, we need to try harder to support `"best"` in local training mode too.

Reply:
This might be worth resolving after dogfooding...